src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor_Minimal.cs

```diff
@@ -39,19 +37,7 @@ protected bool NameBoundSuccessfullyToSameSymbol(INamedTypeSymbol symbol)
             ImmutableArray<ISymbol> typeOnlySymbols = semanticModelOpt.LookupNamespacesAndTypes(positionOpt, name: symbol.Name);
             ISymbol typeOnlySymbol = SingleSymbolWithArity(typeOnlySymbols, symbol.Arity);
 
-            if (typeOnlySymbol == null)
-            {
-                return false;
-            }
-
-            var type1 = GetSymbolType(normalSymbol);
-            var type2 = GetSymbolType(typeOnlySymbol);
-
-            return
-                type1 != null &&
-                type2 != null &&
-                type1.Equals(type2) &&
-                typeOnlySymbol.Equals(symbol.OriginalDefinition);
+            return typeOnlySymbol == null ? false : typeOnlySymbol.Equals(symbol.OriginalDefinition);
```

Comment:
>return typeOnlySymbol == null ? false : typeOnlySymbol.Equals(symbol.OriginalDefinition); [](start = 12, length = 89)

Given this change, we are no longer targeting the "Color Color" case as the comment above implies. At the very least the comment should be adjusted to explain why this logic is sound. To me, it looks like this new logic is valid only if we know that we are in a type-only context, which we cannot be sure of, given the limited information this function has. It feels like we should do more testing and verify that we are not breaking behavior in a non-type-only context.

Reply:
@cston Why do we need namespace for return value here? It looks like a bug, isn't it?
It make sense in this case:
```csharp
using C1;
using C2;

namespace ConsoleApplication1
{
    class Program
    {
        static C2.A B { get; set; }
        static B A { get; set; }
        static C C { get; set; }

        static void Main(string[] args)
        {
            var a = new C1.A();
            B = C2.B$$.F();
        }
    }   
}

namespace C1
{
    public class A { }

    public class C { }
}

namespace C2
{
    public class A { }

    class B
    {
        public static A F() => null;
    }
}
```

Reply:
@zaytsev-victor, yes, it looks like the namespace qualifier is unnecessary for the return type in the `PropertyWithSameNameAsOtherType` test above.

Reply:
@cston Could you please explain what is "Color Color" case?) Because I'm starting to think that simplifying of ```NameBoundSuccessfullyToSameSymbol``` function was the correct fix.

Reply:
The "Color Color" case is the where the member name and the member type name are identical: see [C# spec](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#identical-simple-names-and-type-names). Unfortunately simplifying `NameBoundSuccessfullyToSameSymbol` will also simplify cases where the member name matches the type name of a different member which is not correct: see `PropertyWithSameNameAsOtherType` test above.

Reply:
@cston but you are agree that the namespace qualifier is unnecessary for the return type in the ```PropertyWithSameNameAsOtherType``` test. If we really want (for some reason) the namespace qualifier in quick info in these cases it is better to create new format option for quick info, isn't it? Because the namespace qualifier isn't required in code.

Reply:
I think it's reasonable to have the namespace qualifier on both the return type and member if the namespace is required for one of the two.